cds-plugin.js

```diff
@@ -210,17 +223,23 @@ module.exports = class AdvancedEventMesh extends cds.MessagingService {
     })
 
     const _scheduleUpdateToken = () => {
-      const waitingTime = (Math.max(this.token_expires_in - 10, 0)) * 1000
+      const waitingTime = Math.max(this.token_expires_in - 10, 0) * 1000
       setTimeout(async () => {
         this.LOG._info && this.LOG.info('Fetching fresh token')
-        await _fetchToken()
-        this.session.updateAuthenticationOnReconnect({ accessToken: this.token })
-        _scheduleUpdateToken()
+        try {
+          const { access_token, expires_in } = await _fetchToken(auth_srv)
+          this.token = access_token
+          this.token_expires_in = expires_in
+          this.session.updateAuthenticationOnReconnect({ accessToken: this.token })
+          _scheduleUpdateToken()
+        } catch (error) {
+          // TODO: What to do here?
```

Comment:
@David-Kunz what should we do here? terminate session? then we'd need lazy sessions...